DataFormats/Detectors/CTP/src/Scalers.cxx

```diff
@@ -674,35 +674,35 @@ std::pair<double, double> CTPRunScalers::getRate(uint32_t orbit, int classindex,
 
   // then we can use binary search to find the right entries
   auto iter = std::lower_bound(mScalerRecordO2.begin(), mScalerRecordO2.end(), orbit, [&](CTPScalerRecordO2 const& a, uint32_t value) { return a.intRecord.orbit <= value; });
-  auto nextindex = iter - mScalerRecordO2.begin(); // this points to the first index that has orbit greater or equal to given orbit
+  auto nextindex = std::distance(mScalerRecordO2.begin(), iter); // this points to the first index that has orbit greater or equal to given orbit
 
   auto calcRate = [&](auto index1, auto index2) -> double {
-    auto next = &mScalerRecordO2[index2];
-    auto prev = &mScalerRecordO2[index1];
-    auto timedelta = (next->intRecord.orbit - prev->intRecord.orbit) * 88.e-6; // converts orbits into time
+    const auto& snext = mScalerRecordO2[index2];
+    const auto& sprev = mScalerRecordO2[index1];
+    auto timedelta = (snext.intRecord.orbit - sprev.intRecord.orbit) * 88.e-6; // converts orbits into time
     if (type < 7) {
-      auto s0 = &(prev->scalers[classindex]); // type CTPScalerO2*
-      auto s1 = &(next->scalers[classindex]);
+      const auto& s0 = sprev.scalers[classindex]; // type CTPScalerO2*
+      const auto& s1 = snext.scalers[classindex];
       switch (type) {
         case 1:
-          return (s1->lmBefore - s0->lmBefore) / timedelta;
+          return (s1.lmBefore - s0.lmBefore) / timedelta;
         case 2:
-          return (s1->lmAfter - s0->lmAfter) / timedelta;
+          return (s1.lmAfter - s0.lmAfter) / timedelta;
         case 3:
-          return (s1->l0Before - s0->l0Before) / timedelta;
+          return (s1.l0Before - s0.l0Before) / timedelta;
         case 4:
-          return (s1->l0After - s0->l0After) / timedelta;
+          return (s1.l0After - s0.l0After) / timedelta;
         case 5:
-          return (s1->l1Before - s0->l1Before) / timedelta;
+          return (s1.l1Before - s0.l1Before) / timedelta;
         case 6:
-          return (s1->l1After - s0->l1After) / timedelta;
+          return (s1.l1After - s0.l1After) / timedelta;
         default:
           LOG(error) << "Wrong type:" << type;
           return -1; // wrong type
       }
     } else if (type == 7) {
-      auto s0 = &(prev->scalersInps[classindex]); // type CTPScalerO2*
-      auto s1 = &(next->scalersInps[classindex]);
+      auto s0 = sprev.scalersInps[classindex]; // type CTPScalerO2*
+      auto s1 = snext.scalersInps[classindex];
```

Comment:
@lietava the real bug was here (the address of the counter was taken instead of its value) the rest is cosmetics.